src/source/video_source.js

```diff
@@ -107,7 +107,7 @@ class VideoSource extends ImageSource {
     // setCoordinates inherited from ImageSource
 
     prepare() {
-        if (!this.tile || this.video.readyState < 2) return; // not enough data for current position
+        if (Object.keys(this.tiles).length === 0 || this.video.readyState < 2) return; // not enough data for current position
```

Comment:
I updated `CanvasSource`, but I didn't see an equivalent of `debug/image.html` for testing it -- do we have any working examples?